src/lib/input/input-container.ts

```diff
@@ -267,7 +268,8 @@ export class MdInputDirective {
   /** Determines if the component host is a textarea. If not recognizable it returns false. */
   private _isTextarea() {
     let nativeElement = this._elementRef.nativeElement;
-    return nativeElement ? nativeElement.nodeName.toLowerCase() === 'textarea' : false;
+    let nodeName = this._platform.isBrowser ? nativeElement.nodeName : nativeElement.name;
```

Comment:
Should add a comment that explains why this falls back to name
(also shouldn't be necessary long-term since platform-server is changing to mimic more DOM APIs)